php/WP_CLI/Process.php

```diff
@@ -112,6 +112,10 @@ public function run_check() {
 		if ( $r->return_code || ! empty( $r->STDERR ) ) {
 			throw new \RuntimeException( $r );
 		}
+		// But do it right if testing.
+		if ( ! empty( $this->env['BEHAT_RUN'] ) && ! empty( $r->stderr ) ) {
```

Comment:
Can we perform this check where `run_check()` is called, instead of modifying the `Process` class itself?

Reply:
It's called in a lot of places - do mean just in `invoke_proc()` https://github.com/wp-cli/wp-cli/blob/v1.4.1/features/steps/when.php#L7?

Reply:
Yes. Given this change is specific to the test suite, it makes more sense to include it in the test suite.

Reply:
Hmm, that would make the code a bit yuck, which usually isn't a good sign. Is the `BEHAT_RUN` check not exclusive enough?

Reply:
> Hmm, that would make the code a bit yuck, which usually isn't a good sign.

Why?

> Is the `BEHAT_RUN` check not exclusive enough?

It's hacky. It's not the Process class' responsibility to adapt behavior based on some environmental flag. It's Behat that wants to change the behavior of `STDERR`.

Reply:
I'll add it as a new function `run_check_stderr()` - that should keep us both happy...